homeassistant/components/sensor/recorder.py

```diff
@@ -637,35 +637,70 @@ def validate_statistics(
     """Validate statistics."""
     validation_result = defaultdict(list)
 
-    sensor_states = _get_sensor_states(hass)
+    sensor_states = hass.states.all(DOMAIN)
 
     for state in sensor_states:
         entity_id = state.entity_id
         device_class = state.attributes.get(ATTR_DEVICE_CLASS)
+        state_class = state.attributes.get(ATTR_STATE_CLASS)
         state_unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
 
-        if device_class not in UNIT_CONVERSIONS:
-            metadata = statistics.get_metadata(hass, (entity_id,))
-            if not metadata:
-                continue
+        metadata = statistics.get_metadata(hass, (entity_id,))
```

Comment:
Let's move this to do 1 lookup for all metadata instead of 1 query per entity (which can run into the 100s or 1000s depending on installation)